packages/autorest.python/autorest/codegen/models/base.py

```diff
@@ -45,6 +45,8 @@ def __init__(self, yaml_data: Dict[str, Any], code_model: "CodeModel") -> None:
         self.api_versions: List[str] = yaml_data.get(
             "apiVersions", []
         )  # api versions this type is in.
+        self.enable_generate: bool = yaml_data.get("enableGenerate", True)
+        self.enable_overload_check: bool = yaml_data.get("enableOverloadCheck", True)
```

Comment:
when do we disable overload checks?

Reply:
we generate free overload `IO` for `bytes` but we usually check `(IO, bytes)` together so we disable overload check for bytes:https://github.com/Azure/autorest.python/blob/00bc12a382db3cd3cd33b97c5c815c8a347312d2/packages/typespec-python/test/generated/overload/overload/_operations/_operations.py#L74-L161